### ISSUE TITLE
instanceOf and Mockito mock

### ISSUE THREAD

**Issue by jaspervp** [google/j2objc#879]
When you create a mock with Mockito, the mocked class is not a superclass of the mocked object.

This tests works on java, but not after being transpiled.

```import org.junit.Test;

import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.mock
public class InstanceOfTest {

    @Test
    public void test()
    {

        Object test = mock(InstanceOfTest.class);
        Class C = test.getClass();
        while (C != null) {
            System.out.println(C.getName());
            C = C.getSuperclass();
        }
        boolean result = (test instanceof InstanceOfTest);
        assertTrue(result);
    }
}
```

**Comment by kingxt** [comment]
I also found the problem when build with ios11, but run well on ios11 before

**Comment by sfuku7** [comment]
Hi

I created PR #922 to fix the issue.
Could you please review that ?

**Comment by tomball** [comment]
Fixed in current source via PR #922.
